library/kotlin/test/io/envoyproxy/envoymobile/GRPCStreamEmitterTest.kt

```diff
@@ -0,0 +1,109 @@
+package io.envoyproxy.envoymobile
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.io.ByteArrayOutputStream
+import java.nio.ByteBuffer
+
+class GRPCStreamEmitterTest {
+
+  // TODO: Problems with nhaarman/mockito-kotlin https://github.com/lyft/envoy-mobile/issues/504
+  // This is a total hack to get something to work
+  private lateinit var emitter: StreamEmitter
+
+  private val dataOutputStream = ByteArrayOutputStream()
+  private val closeOutputStream = ByteArrayOutputStream()
+  private var isCloseCalled = false
+
+  @Before
+  fun setup() {
+    emitter = object : StreamEmitter {
+
+      override fun cancel() {
+        TODO("not implemented")
+      }
+
+      override fun sendData(byteBuffer: ByteBuffer): StreamEmitter {
+        dataOutputStream.write(byteBuffer.array())
+        return this
+      }
+
+      override fun sendMetadata(metadata: Map<String, List<String>>): StreamEmitter {
+        TODO("not implemented")
```

Comment:
So a `TODO` actually causes an error if the code ever gets executed -- This wasn't a fun test to write but since we have https://github.com/lyft/envoy-mobile/issues/504, I decided to just keep it this way -- we can alternatively just throw an exception if you feel that aesthetic is better

Reply:
Hm yea I think throwing an exception might be better here. My thought was to reduce false-positives when grepping the codebase for "todo" items